images/ipfailover/keepalived/lib/config-generators.sh

```diff
@@ -214,15 +182,27 @@ function generate_vrrpd_instance_config() {
 
   local vipname=$(scrub "$1")
   local initialstate=""
-  local preempt=${PREEMPTION:-"$DEFAULT_PREEMPTION_STRATEGY"}
+  local preempt=${PREEMPTION:-"${DEFAULT_PREEMPTION_STRATEGY}"}
   local vrrpidoffset=${HA_VRRP_ID_OFFSET:-0}
 
-  [ "$instancetype" = "master" ] && initialstate="state MASTER"
+  [ "${instancetype}" = "master" ] && initialstate="state MASTER"
 
-  local instance_name=$(generate_vrrp_instance_name "$servicename" "$iid")
+  local instance_name ; instance_name=$(generate_vrrp_instance_name "${servicename}" "${iid}")
```

Comment:
The comment here likewise applies to every line with a scoping statement (`local`, `export`, `readonly`, *etc*) and a command substitution or other construct that has a return code. You didn't add these, so if you want to defer until another PR to fix them all, that's probably fine.